src/Chainweb/Pact/PactService.hs

```diff
@@ -502,6 +508,13 @@ execNewBlock mpAccess parent miner = do
         blockGasLimit <- view psBlockGasLimit
         let initState = BlockFill blockGasLimit mempty 0
 
+        let 
```

Comment:
if you move this to function-toplevel `where` then you won't have to thread it into `refill` etc, since everything in `execNewBlock` uses this

Reply:
This uses the `blockGasLimit` which we fetch from the reader env, so it can't move all the way out to the function level.

Reply:
ah crap. ok ... we need to clean up this whole refill mess anyway so we can refactor then.